src/services/connextService.ts

```diff
@@ -0,0 +1,96 @@
+import { getPoolFeeForUniV3, getXCallCallData, prepareSwapAndXCall } from '@connext/chain-abstraction';
+import { DestinationCallDataParams, Swapper, SwapAndXCallParams } from '@connext/chain-abstraction/dist/types';
+import { SdkConfig, create } from '@connext/sdk';
+import { RAW_NETWORKS, SUPPORTED_CHAINS_BY_CONNEXT } from 'config';
+import WalletService from './walletService';
+
+interface DomainID {
+  [key: number]: string;
+}
+
+export default class ConnextService {
+  sdkConfig: SdkConfig;
+
+  walletService: WalletService;
+
+  destinantionChainID: number;
+
+  constructor(walletService: WalletService, destinantionChainID: number) {
+    this.walletService = walletService;
+    this.destinantionChainID = destinantionChainID;
+  }
+
+  getRPCURL(chainID: number) {
+    const network = Object.values(RAW_NETWORKS).find((net) => net.chainId === chainID);
+    return network?.rpc[0];
+  }
+
+  async getCalculatedRelayerFees(originDomain: string, destinationDomain: string) {
+    const domainConfig: { [domainId: string]: { providers: string[] } } = {};
+
+    const domainChainIds = Object.entries(SUPPORTED_CHAINS_BY_CONNEXT)
+      .filter(([key]) => typeof key === 'number')
+      .map(([key, value]) => ({ domainId: value.domainId, chainId: key }));
+
+    domainChainIds.forEach((obj) => {
+      domainConfig[obj.domainId] = { providers: [this.getRPCURL(parseInt(obj.chainId, 10)) as string] };
+    });
+
+    this.sdkConfig = {
+      signerAddress: this.walletService.account as string,
+      network: 'mainnet', // can change it to testnet as well
+      chains: domainConfig,
+    };
+    const { sdkBase } = await create(this.sdkConfig);
+    const relayerFees = await sdkBase.estimateRelayerFee({
+      originDomain,
+      destinationDomain,
+      isHighPriority: true,
+    });
+    return relayerFees.toString();
+  }
+
+  async getPoolFeeForUniV3Helper(domainID: string, token0: string, token1: string, rpcURL: string) {
+    if (!rpcURL || !token0 || !token1) {
+      return null;
+    }
+    const poolFee = await getPoolFeeForUniV3(domainID, rpcURL, token0, token1);
+    return poolFee;
+  }
+
+  async getXCallCallDataHelper(domainID: string, forwardCallData: string, params: DestinationCallDataParams) {
+    const swapper = Swapper.UniV3;
+    const callDataForMeantTarget = await getXCallCallData(domainID, swapper, forwardCallData, params);
+    return callDataForMeantTarget;
+  }
+
+  async prepareSwapAndXCallHelper(swapAndXCallParams: SwapAndXCallParams, signerAddress: string) {
+    const txRequest = await prepareSwapAndXCall(swapAndXCallParams, signerAddress);
+    return txRequest;
+  }
+
+  getDomainID(networkName: number): string {
+    const domainID: DomainID = {
```

Comment:
Isn't this already on SUPPERTED_CHAINS_BY_CONNEXT?

Reply:
Adding it as a class method so that it can be fetched by connextService itself, will move the supported chains by connext here as well

Reply:
bump to this